src/main/java/minijava/lexer/LexerInput.java

```diff
@@ -0,0 +1,23 @@
+package minijava.lexer;
+
+import java.util.Iterator;
+
+/** Input for a lexer (a stream of integers) it stops after the EOF character is read. */
+interface LexerInput extends Iterator<Integer> {
```

Comment:
I'm currently fuzzing the lexer. If the lexer doesn't yield to errors, then I commit it.